### ISSUE TITLE
Unable to fetch user's profile image

### ISSUE THREAD

**Issue by ovidiucp** [microsoftgraph/microsoft-graph-docs#3990]
Issue: 

MS Graph version 1.0 is unable to fetch a user's profile image

Article:

The document at

https://docs.microsoft.com/en-us/graph/api/profilephoto-get?view=graph-rest-1.0

mentions that v1.0 of the API is not able to retrieve a user's profile image:

    Note This operation in version 1.0 supports only a user's work or school mailboxes 
    and not personal mailboxes.

FYI, I make a request to

`GET https://graph.microsoft.com/v1.0/me/photo/$value`

And the error I receive is:

```
Error Domain: com.microsoft.graph.errors
Code=501"unimplemented"
```

This is a basic feature that all other major OAuth providers implement. When should we expect this feature to work?

* ID: b7710698-d997-c4c0-8515-dfb72d3551af
* Version Independent ID: 81ccdb02-f40d-277d-d51e-157d654a6e3a
* Content: [Get photo - Microsoft Graph v1.0](https://docs.microsoft.com/en-us/graph/api/profilephoto-get?view=graph-rest-1.0)
* Content Source: [api-reference/v1.0/api/profilephoto-get.md](https://github.com/microsoftgraph/microsoft-graph-docs/blob/master/api-reference/v1.0/api/profilephoto-get.md)
* Product: **non-product-specific**
* GitHub Login: @MSGraphDocsvTeam
* Microsoft Alias: **MSGraphDocsVteam**

**Comment by kevinbellinger** [comment]
We're in the process of enabling this feature on the V1.0 endpoint at present. 

Historically access to these images was provided in a different manner which presented challenges that needed addressing first - we are finalizing that work currently, once completed we'll be able to finalize.

**Comment by ovidiucp** [comment]
Excellent Kevin, thanks for doing this! I look forward to enabling this feature in our [mobile apps](https://jollyturns.com).

**Comment by DingmaomaoBJTU** [comment]
Hi@kevinbellinger, is there any update on this?

**Comment by DingmaomaoBJTU** [comment]
> We're in the process of enabling this feature on the V1.0 endpoint at present.
> 
> Historically access to these images was provided in a different manner which presented challenges that needed addressing first - we are finalizing that work currently, once completed we'll be able to finalize.

Hi@kevinbellinger, is there any update on this?

**Comment by linhntaim** [comment]
Please support for personal mailboxes. Thanks!

**Comment by kevinbellinger** [comment]
We're actively working on this becoming possible. Our hope is to release in the next 6-8weeks.

**Comment by arnemo** [comment]
Any update on when this feature will be available? Does the feature require the user to have an Office365 subscription?

**Comment by mohsinhere** [comment]
Any updates on this yet?

For me,
Works: https://graph.microsoft.com/beta/users/{id}/photo/$value
Doesnt work: https://graph.microsoft.com/v1.0/users/{id}/photo/$value

**Comment by CelesteDG** [comment]
Flagging @kevinbellinger for a follow up response. Hi @kevinbellinger - Can you please confirm if this feature is now available and whether an O365 or M365 subscription is required for users' profile images to show up? Also please see the above question from mohsinhere. It seems contradictory to the info from 3293.

**Comment by kevinbellinger** [comment]
We ran into some challenges with promoting this feature for MSA accounts. I'll come back with an ETA once we address the concerns.

**Comment by mishas2303** [comment]
Any ETA on this yet?

**Comment by veler** [comment]
Hello, any ETA?

**Comment by ButtiBBQ** [comment]
Here is the same...

works: https://graph.microsoft.com/beta/me/photo/$value
Doesnt work: https://graph.microsoft.com/v1.0/me/photo/$value

Any ETA for a fix?

**Comment by yashvit** [comment]
> Any updates on this yet?
> 
> For me,
> Works: [https://graph.microsoft.com/beta/users/{id}/photo/$value](https://graph.microsoft.com/beta/users/%7Bid%7D/photo/$value)
> Doesnt work: [https://graph.microsoft.com/v1.0/users/{id}/photo/$value](https://graph.microsoft.com/v1.0/users/%7Bid%7D/photo/$value)

Any updates on this?

I too cant manage to fetch a user profile pic nor update it:
GET / PUT https://graph.microsoft.com/v1.0/users/{id}/photo/$value

Have the necessary permissions mentioned in the docs, but still get the following error:

```
"code": "NoPermissionsInAccessToken",
"message": "The token contains no permissions, or permissions can not be understood."
```

**Comment by bzmo** [comment]
Any updates on this? I'm interested in retrieving a profile photo and its metadata for MSA accounts.

Though the beta version supports retrieving the profile photo, there's no support (as far as I can tell) for retrieving a photo and its metadata of a specific smaller size and no `etag` property for MSA accounts.

**Comment by yaw-innovate** [comment]
Hi, this is available in our service backlog and it is getting the needed attention. I will come back with an ETA when the underlying concerns are addressed.

**Comment by Neutrino-Sunset** [comment]
Two and a half years later, still doesn't work.

**Comment by isvargasmsft** [comment]
Hello, 

Our deepest apologies for the delays in our response. 

The final fix has been delayed as we discovered further design challenges to resolve before fixing this issue. My sincere apologies as I understand we provided incorrect ETAs previously. 

Unfortunately, since this fix requires a deeper work, we can't provide an ETA yet. 

We have this issue noted and we'll keep you posted.

Best regards, 

Isaac Vargas

**Comment by Neutrino-Sunset** [comment]
Are you aware that this doesn't even work properly in the beta api? In the beta api it works if accessed using an access token for a multi-tenant application but not for a single tenant application. As described here:

https://github.com/microsoftgraph/microsoft-graph-explorer-v4/issues/1077

That said it *does* work in the beta api for a multi-tenant application, so I don't see how a delay of this magnitude can be justified on the basis of "design challenges". After all you already have one functioning implementation, how difficult can it be to access that functioning implementation for your non-functioning one?

**Comment by yaw-innovate** [comment]
Hi, thank you all for your patience while we worked on this.
The consumer (MSA) profile photo is now available on MS Graph v1.0 - this is valid for owner (signed-in user) scenarios with the permission scopes User.Read, User.ReadWrite 

The documentation has been updated accordingly https://docs.microsoft.com/en-us/graph/api/profilephoto-get?view=graph-rest-1.0

Thanks again!

**Comment by Neutrino-Sunset** [comment]
The dcumentation page still states

> Note This operation in version 1.0 supports only a user's work or school mailboxes and not personal mailboxes.

Although I just tested it using the Graph Explorer and the api now appears to work for personal accounts too.
